src/test/javascript/spec/component/tutor-dashboard/tutor-exercise-dashboard.spec.ts

```diff
@@ -0,0 +1,120 @@
+import { MockRouter } from '../../mocks/mock-router.service';
+import { MockTutorParticipationService } from '../../mocks/mock-tutor-participation.service';
+import { MockTextSubmissionService } from '../../mocks/mock-text-submission.service';
+import { MockMarkdownService } from '../../mocks/mock-markdown.service';
+import { MockComplaintService } from '../../mocks/mock-complaint.service';
+import { ArTEMiSHeaderExercisePageWithDetailsModule } from 'app/exercise-headers';
+import { ArTEMiSTutorCourseDashboardModule } from 'app/tutor-course-dashboard';
+import { ArTEMiSHomeModule } from 'app/home';
+import { ArTEMiSCourseModule } from 'app/entities/course/course.module';
+import { ArTEMiSSidePanelModule } from 'app/components/side-panel/side-panel.module';
+import { ArTEMiSModelingEditorModule } from 'app/modeling-editor';
+import { AssessmentInstructionsModule } from 'app/assessment-instructions/assessment-instructions.module';
+import { TutorParticipationStatus } from 'app/entities/tutor-participation';
+import { of, throwError } from 'rxjs';
+import { HttpErrorResponse } from '@angular/common/http';
+import { ArTEMiSMainModule } from 'app/layouts/main/main.module';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('TutorExerciseDashboardComponent', () => {
+    let comp: TutorExerciseDashboardComponent;
+    let fixture: ComponentFixture<TutorExerciseDashboardComponent>;
+    let modelingSubmissionService: ModelingSubmissionService;
+
+    const exercise = { id: 20, type: ExerciseType.MODELING, tutorParticipations: [{ status: TutorParticipationStatus.TRAINED }] } as ModelingExercise;
+    const submission = { id: 30 } as ModelingSubmission;
+    const lockLimitErrorResponse = new HttpErrorResponse({ error: { errorKey: 'lockedSubmissionsLimitReached' } });
+
+    beforeEach(async () => {
+        return TestBed.configureTestingModule({
+            imports: [
+                ArTEMiSTestModule,
+                ArTEMiSSharedModule,
+                ArTEMiSHeaderExercisePageWithDetailsModule,
+                ArTEMiSTutorCourseDashboardModule,
+                ArTEMiSHomeModule,
+                ArTEMiSCourseModule,
+                ArTEMiSMainModule,
+                ArTEMiSSidePanelModule,
+                ArTEMiSModelingEditorModule,
+                ArTEMiSResultModule,
+                AssessmentInstructionsModule,
+                RouterModule,
+            ],
+            declarations: [TutorExerciseDashboardComponent],
+            providers: [
+                JhiLanguageHelper,
+                { provide: JhiAlertService, useClass: MockAlertService },
+                { provide: ActivatedRoute, useClass: MockActivatedRoute },
+                { provide: Router, useClass: MockRouter },
+                { provide: TutorParticipationService, useClass: MockTutorParticipationService },
+                { provide: TextSubmissionService, useClass: MockTextSubmissionService },
```

Comment:
I think we could remove the four providers above as there is no interaction with these services in the tests. What do you think @thilo-behnke?

Reply:
You only have to add the services that are strictly needed for your tests in my opinion. So if service foo is never called, don't bother to add it yet.
I think this was also an issue with the entry components of the module.